pkg/storageincentives/staking/contract.go

```diff
@@ -224,3 +228,74 @@ func (c *contract) getBalance(ctx context.Context) (*big.Int, error) {
 	}
 	return abi.ConvertType(results[0], new(big.Int)).(*big.Int), nil
 }
+
+func (c *contract) WithdrawAllStake(ctx context.Context) (txHash common.Hash, err error) {
+	isPaused, err := c.paused(ctx)
+	if err != nil {
+		return
+	}
+	if !isPaused {
+		err = ErrNotPaused
+		return
+	}
+
+	stakedAmount, err := c.getStake(ctx, c.overlay)
+	if err != nil {
+		return
+	}
+
+	if stakedAmount.Cmp(big.NewInt(0)) <= 0 {
+		err = ErrInsufficientStake
+		return
+	}
+
+	_, err = c.sendApproveTransaction(ctx, stakedAmount)
+	if err != nil {
+		return
+	}
+
+	receipt, err := c.withdrawFromStake(ctx, stakedAmount)
+	if receipt != nil {
+		txHash = receipt.TxHash
+	}
+	return
+}
+
+func (c *contract) withdrawFromStake(ctx context.Context, stakedAmount *big.Int) (*types.Receipt, error) {
+	var overlayAddr [32]byte
+	copy(overlayAddr[:], c.overlay.Bytes())
```

Comment:
This is not needed, you can use `c.overlay.Clone().Bytes()` instead.

Reply:
I think the abi packing needs a [32]byte and not a byte slice